### PR TITLE
HPCC-22594 Hex Preview fails to render

### PR DESCRIPTION
**Pull request by GordonSmith**
Replaced WU creation to use @hpcc-js/comms

Signed-off-by: Gordon Smith <gordonjsmith@gmail.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by GordonSmith** [comment]
@miguelvazq please review

File: esp/src/eclwatch/HexViewWidget.js [hunk 1]
```diff
@@ -87,74 +84,45 @@ define([
...
+                                msg += exception.Message;
                             }
-                            break;
-                        case "State":
-                            context.hexView.setText("..." + (context.wu.isComplete() ? context.i18n.fetchingresults : newValue) + "...");
-                            break;
+                        });
+                        if (msg) {
+                            dojo.publish("hpcc/brToaster", {
+                                Severity: "Error",
```
Comment (miguelvazq) [inline esp/src/eclwatch/HexViewWidget.js]: Minor: Extra space.

File: esp/src/lws.config.js [hunk 2]
```diff
@@ -16,7 +16,7 @@ let rewrite = [
     { from: "/esp/files/esp/logout", to: "http://" + debugServerIP + ":8010/esp/logout" },
     { from: "/ws_elk/*", to: "http://" + debugServerIP + ":8010/ws_elk/$1" },
     { from: "/esp/files/esp/reset_session_timeout", to: "http://" + debugServerIP + ":8010/esp/reset_session_timeout" },
-    { from: "/esp/files/node_modules/@hpcc-js/*/dist/index.min.js", to: "/node_modules/@hpcc-js/$1/dist/index.js" },
+    { from: "/esp/files/node_modules/@hpcc-js/*/dist/index.min.js", to: debugServerIP !== CLUSTER_GJS ? "/node_modules/@hpcc-js/$1/dist/index.js" : "/hpcc-js/$1/dist/index.js" },
```
Comment (miguelvazq) [inline esp/src/lws.config.js]: Won't this interfere with other developers configurations if I do not have `CLUSTER_GJS` set?

File: esp/src/eclwatch/HexViewWidget.js [hunk 3]
```diff
@@ -87,74 +84,45 @@ define([
...
+                                Source: "HexViewWidget.remoteRead",
+                                Exceptions: [{ Source: context.wu.Wuid, Message: msg }]
+                            });
+                        }
                     }
+                    return wu.fetchResults().then(function (results) {
+                        return results.length ? results[0].fetchRows() : [];
+                    }).then(function (rows) {
+                        context.cachedResponse = "";
```
Comment (miguelvazq) [inline esp/src/eclwatch/HexViewWidget.js]: Not sure why the reassignment of type is happening here unless rows is a string as well?

**Review by miguelvazq — CHANGES_REQUESTED** [review]

File: esp/src/lws.config.js (hunk 2 above)
Comment (GordonSmith) [reply esp/src/lws.config.js]: Nope - it will default to the old condition.

File: esp/src/eclwatch/HexViewWidget.js (hunk 3 above)
Comment (GordonSmith) [reply esp/src/eclwatch/HexViewWidget.js]: Yup - looks wrong.

Comment (GordonSmith) [reply esp/src/eclwatch/HexViewWidget.js]: Removed spurious line.

File: esp/src/eclwatch/HexViewWidget.js (hunk 1 above)
Comment (GordonSmith) [reply esp/src/eclwatch/HexViewWidget.js]: You sure?

**Comment by GordonSmith** [comment]
@miguelvazq changes made.

File: esp/src/eclwatch/HexViewWidget.js (hunk 1 above)
Comment (miguelvazq) [reply esp/src/eclwatch/HexViewWidget.js]: Never mind. Eyes playing tricks on me.

**Review by miguelvazq — APPROVED** [review]
Looks good now.

**Pull request merged**